src/engine/V3Eval.v3

```diff
@@ -379,6 +424,53 @@ component V3Eval {
 		var r1 = f(a.1, b.1);
 		return (r0, r1);
 	}
+	private def do_vv_v_x4(a: (u64, u64), b: (u64, u64), f: (u32, u32) -> u32) -> (u64, u64) { // Performs a 4-lane binop
+		var r0 = f(u32.view(a.0), u32.view(b.0));
+		var r1 = f(u32.view(a.0 >> 32), u32.view(b.0 >> 32));
+		var r2 = f(u32.view(a.1), u32.view(b.1));
+		var r3 = f(u32.view(a.1 >> 32), u32.view(b.1 >> 32));
+		return ((u64.view(r1) << 32) | r0, (u64.view(r3) << 32) | r2);
+	}
+	private def do_vv_v_x8(a: (u64, u64), b: (u64, u64), f: (u16, u16) -> u16) -> (u64, u64) { // Performs an 8-lane binop
+		var result0: u64 = 0;
+		var result1: u64 = 0;
+
+		for (i < 4) {
+			var shift_amount: byte = byte.!(i * 16);
+			
+			var r_a = u16.view((a.0 >> shift_amount) & 0xFFFF);
+			var r_b = u16.view((b.0 >> shift_amount) & 0xFFFF);
+			var res = f(r_a, r_b);
+			result0 |= (u64.view(res) << shift_amount);
+			
+			r_a = u16.view((a.1 >> shift_amount) & 0xFFFF);
+			r_b = u16.view((b.1 >> shift_amount) & 0xFFFF);
+			res = f(r_a, r_b);
+			result1 |= (u64.view(res) << shift_amount);
+		}
+
+		return (result0, result1);
+	}
+	private def do_vv_v_x16(a: (u64, u64), b: (u64, u64), f: (u8, u8) -> u8) -> (u64, u64) { // Performs a 16-lane binop
+		var result0: u64 = 0;
+		var result1: u64 = 0;
+
+		for (i < 8) {
```

Comment:
You can do `for (shift: byte = 0; shift < 64; shift += 8) {`

Reply:
Right right. Got rid of that i and renamed some locals.